documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc

```diff
@@ -145,4 +145,8 @@ Type: _int_ | false | `1000`
 
 Type: _boolean_ | false | `true`
 
+| *batch* | Whether the Kafka records are consumed in batch
+
+Type: _boolean_ | false | `false`
```

Comment:
Is that something we could derive from the `@Incoming` method parameter type (or `@Channel` injection point type)? Maybe only in Quarkus, like we autodetect the [de]serializer.

Reply:
There will be another PR on Quarkus to adapt the serde autodetect. I was thinking more of a suggestion message rather than automatically setting a channel to batch.

A note here: an incoming channel can not have two consuming `@Incoming`/`@Channel` that are batch and not batch in the same time.

Reply:
Right! Hence I thought that this can be unambiguously detected.

Reply:
> A note here: an incoming channel can not have two consuming @Incoming/@Channel that are batch and not batch in the same time.

I would consider that invalid. The workaround would be to consume it using batch twice and having an application method unwrapping the records. 

About the detection in Quarkus: +1